arcgis-ios-sdk-samples/Maps/Display device location with NMEA data sources/DisplayDeviceLocationWithNMEADataSourcesViewController.swift

```diff
@@ -26,29 +26,90 @@ class DisplayDeviceLocationWithNMEADataSourcesViewController: UIViewController {
     }
     /// The label to display satellites info.
     @IBOutlet var statusLabel: UILabel!
-    /// The button to start the demo.
-    @IBOutlet var startBarButtonItem: UIBarButtonItem!
+    /// The button to choose a data source and start the demo.
+    @IBOutlet var sourceBarButtonItem: UIBarButtonItem!
     /// The button to reset pan mode to "recenter".
     @IBOutlet var recenterBarButtonItem: UIBarButtonItem!
     /// The button to reset the demo.
     @IBOutlet var resetBarButtonItem: UIBarButtonItem!
     
+    // MARK: Constants
+    
+    /// The protocols specified in the `Info.plist` that the app uses to
+    /// communicate with external accessory hardware.
+    let supportedProtocolStrings: [String] = {
+        guard let protocols = Bundle.main.object(forInfoDictionaryKey: "UISupportedExternalAccessoryProtocols") as? [String] else {
+            return []
+        }
+        return protocols
+    }()
+    
     // MARK: Instance properties
     
     /// An NMEA location data source, to parse NMEA data.
-    let nmeaLocationDataSource = AGSNMEALocationDataSource(receiverSpatialReference: .wgs84())
+    var nmeaLocationDataSource: AGSNMEALocationDataSource?
+    /// A string to hold the latest satellite info.
+    var satelliteInfoText = "Satellites info will be shown here."
     /// A mock data source to read NMEA sentences from a local file, and generate
     /// mock NMEA data every fixed amount of time.
     let mockNMEADataSource = SimulatedNMEADataSource(nmeaSourceFile: Bundle.main.url(forResource: "Redlands", withExtension: "nmea")!, speed: 1.5)
+    /// A formatter for the accuracy distance string.
+    let distanceFormatter: MeasurementFormatter = {
+        let formatter = MeasurementFormatter()
+        formatter.unitOptions = .naturalScale
+        formatter.numberFormatter.minimumFractionDigits = 1
+        formatter.numberFormatter.maximumFractionDigits = 1
+        return formatter
+    }()
     
     // MARK: Actions
     
-    @IBAction func start() {
+    @IBAction func chooseDataSource(_ sender: UIBarButtonItem) {
+        let alertController = UIAlertController(
+            title: "Choose an NMEA data source.",
+            message: nil,
+            preferredStyle: .actionSheet
+        )
```

Comment:
Is there a reason why the sample can't use the bluetooth accessory picker ([`EAAccessoryManager.showBluetoothAccessoryPicker(withNameFilter:completion:)`](https://developer.apple.com/documentation/externalaccessory/eaaccessorymanager/1613913-showbluetoothaccessorypicker))? According to the documentation for `EAAccessoryManager.connectedAccessories`, "Because the contents of this property can change dynamically based on the connection and disconnection of accessories, you should not cache the value of this property.", which is kind of what this sample is doing by creating a static alert view. Presumably the picker stays up-to-date as accessories come and go. Also, the documentation for `EAAccessory.protocolStrings` says "When encountering a new accessory, always check the list of supported protocols for the one you expect to use before creating a session. " So according to the documentation, the protocol check should occur after connection but before creating a session, not while discovering accessories.

Reply:
Oh 🤦  I vaguely remember there is a system built-in UI for picking Bluetooth device, but an earlier search yields no result for me (maybe I'm just not using the right keywords). So thanks for pointing this out!

Reply:
@philium Today when I was trying the BT picker with a newly created Xcode 12.5 project, I found it not showing when `SceneDelegate` methods are present. Searched online and someone [mentioned](https://developer.apple.com/forums/thread/128044?answerId=652027022#652027022) `UIAlertView`, which I don't quite understand 🤔 ...

Anyway, it isn't an problem for the Samples app for now, but might worth a revisit in the future.